core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -67,19 +66,20 @@ protected boolean isAvailable(final BigInteger offset, final BigInteger length)
 
     private synchronized byte[] getValue() {
         if (cache == null) {
-            final ImmutableList<Optional<Value>> results = dataExpression.eval(parseState, encoding);
+            final ImmutableList<Value> results = dataExpression.eval(parseState, encoding);
             if (results.size <= index) {
-                throw new IllegalStateException(format("ValueExpression dataExpression yields %d result(s) (expected at least %d).", results.size, index+1));
+                throw new IllegalStateException(format("ValueExpression dataExpression must yield at least %d results.", index+1));
```

Comment:
I have no idea... I will revert it. I do remember changing it around, probably because I was trying to make the `format(...)`-method in `Util` simple by only accepting a single substitution argument or something like that.